internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1191,6 +1332,37 @@ func getObservabilityConfig(propertyName string, d *schema.ResourceData) (*vault
 	return getValidObservabilityConfig(config)
 }
 
+// if http observability information is provided, this function ensures that authentication fields are valid and returns the authentication method used
+func validateHTTPAuth(httpBasicUser string, httpBasicPassword string, httpBearerToken string) (diag.Diagnostics, *vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth, *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth) {
+	var httpConfigError diag.Diagnostics
+
+	// only one of basic or bearer authentication should be submitted
+	if httpBearerToken != "" && httpBasicUser != "" || httpBearerToken != "" && httpBasicPassword != "" {
+		httpConfigError = diag.Errorf("http configuration is invalid: either the basic or bearer authentication method can be submitted, but not both")
+	} else if httpBasicUser != "" && httpBasicPassword == "" || httpBasicUser == "" && httpBasicPassword != "" {
+		// http basic requires both the username and password to be filled
+		httpConfigError = diag.Errorf("http configuration is invalid: basic authentication requires username and password")
+	}
+
+	if httpConfigError != nil {
+		return httpConfigError, nil, nil
```

Comment:
isn't this equivalent to returning in each line where you're setting `httpConfigError`?